src/utils/__tests__/ranges.test.ts

```diff
@@ -510,4 +509,263 @@ describe("utils - ranges", () => {
+      ]);
+      expect(insertInto([
+        { start: 0, end: 9 },
+      ], { start: 10, end: 10 })).to.eql([
+        { start: 0, end: 9 },
+      ]);
+    });
+
+    it("should just add the range if we had no original ranges", () => {
+      expect(insertInto([], { start: 0, end: 5 })).to.eql([{ start: 0, end: 5 }]);
+    });
+
+    it("should not add a range contained entirely in a previous one", () => {
+      expect(insertInto([
+        { start: 0, end: 100 },
+      ], { start: 10, end: 20 })).to.eql([
+        { start: 0, end: 100 },
+      ]);
+      expect(insertInto([
+        { start: 0, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: 0, end: 100 })).to.eql([
+        { start: 0, end: 100 },
+        { start: 101, end: 201 },
+      ]);
+    });
+
+    it("should merge a range contained partially in a previous one", () => {
+      expect(insertInto([
+        { start: 0, end: 100 },
+      ], { start: 10, end: 110 })).to.eql([
+        { start: 0, end: 110 },
+      ]);
+      expect(insertInto([
+        { start: 0, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: 90, end: 150 })).to.eql([
+        { start: 0, end: 201 },
+      ]);
+    });
+
+    it("should merge a range contiguous with a previous one", () => {
+      expect(insertInto([
+        { start: 0, end: 100 },
+      ], { start: 100, end: 110 })).to.eql([
+        { start: 0, end: 110 },
+      ]);
+      expect(insertInto([
+        { start: 50, end: 70 },
+      ], { start: 0, end: 50 })).to.eql([
+        { start: 0, end: 70 },
+      ]);
+      expect(insertInto([
+        { start: 0, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: 100, end: 101 })).to.eql([
+        { start: 0, end: 201 },
+      ]);
+    });
+
+    it("should allow a small delta when calculating contiguity", () => {
+      const delta = 1 / 60;
+      expect(insertInto([
+        { start: 0, end: 100 },
+      ], { start: delta + 100, end: 110 })).to.eql([
+        { start: 0, end: 110 },
+      ]);
+      expect(insertInto([
+        { start: 0, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: delta + 100, end: 101 - delta })).to.eql([
+        { start: 0, end: 201 },
+      ]);
+    });
+
+    it("should add a range strictly before what we already have", () => {
+      expect(insertInto([
+        { start: 50, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: 0, end: 10 })).to.eql([
+        { start: 0, end: 10 },
+        { start: 50, end: 100 },
+        { start: 101, end: 201 },
+      ]);
+    });
+
+    it("should add a range strictly after what we already have", () => {
+      expect(insertInto([
+        { start: 50, end: 100 },
+        { start: 101, end: 201 },
+      ], { start: 500, end: 510 })).to.eql([
+        { start: 50, end: 100 },
+        { start: 101, end: 201 },
+        { start: 500, end: 510 },
+      ]);
+    });
+
+    it("should add a range between ranges we already have", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 738. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=435&issues=AWgElM4F1_eUtAp8AFEE&open=AWgElM4F1_eUtAp8AFEE)

[ISSUE_KEY]: AWgElM4F1_eUtAp8AFEE